### PR TITLE
Add docs section to test plan

### PR DESCRIPTION
**Pull request by r0mant**
There were some hiccups with documentation during Teleport 11 release where some things got missed. This PR adds docs section to the test plan, to be owned and performed by the technical writers team.

@ptgott @alexfornuto @rsyracuse Please take a look and let me know if this makes sense.

@stevenGravy Can you also take a look and see if I didn't miss anything? I know you had some feedback about this.

### CODE REVIEW

**Review by alexfornuto — APPROVED** [review]
LGTM, but I defer to those who've been here longer and may know something left out that I don't.

File: .github/ISSUE_TEMPLATE/testplan.md
Comment (avatus) [inline .github/ISSUE_TEMPLATE/testplan.md:1591]:
```suggestion
- [ ] Verify deprecated Teleport versions are added to the older versions page
```

Once [this is merged ](https://github.com/gravitational/docs/pull/182)(or even without it), we should also check to make sure whatever deprecated version is added to the 'older-versions' list. If we move to a version-agnostic page like the one in the PR, we can make this change in the same PR we add the new branch to the config. Maybe we even move the "upcoming releases" to a separate page as well?

**Review by avatus — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/testplan.md
Comment (ptgott) [inline .github/ISSUE_TEMPLATE/testplan.md:1594]:
```suggestion
- [ ] Update the `gravitational/docs` version configuration
  - [ ] Update the version switcher config in `gravitational/docs/config.json`
  - [ ] Update `gravitational/docs/.gitmodules`
- [ ] Ensure that `CHANGELOG.md` is up to date and complete for the default docs version
- [ ] Ensure that the "Upcoming Releases" page only exists for the default docs version
```

The last item assumes that we are taking this approach with the "Upcoming Releases" page: https://github.com/gravitational/teleport/pull/18049

**Comment by zmb3** [comment]
@ptgott @r0mant let's try to get this merged before we cut branch/v12.

File: .github/ISSUE_TEMPLATE/testplan.md
Comment (r0mant) [reply .github/ISSUE_TEMPLATE/testplan.md:1721]: @ptgott I've addressed your feedback.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
